File: C++/Model/pura.cpp
#include "pura.h"

pura::pura(const polinomio & a, const polinomio & b):
    secondoGrado(a,b){
    polinomio temp = unifica();
    temp.ordina();
    temp.eliminaZeri();
    if(temp.size()==2){
        if(temp.getMonomio_pos(0).getGrado()!=razionale(0,1))
            throw erroreGrado();
        if(temp.getMonomio_pos(1).getGrado()!=razionale(2,1))
            throw erroreGrado();
    }
    else
        if(temp.getMonomio_pos(0).getGrado()!=razionale(2,1))
            throw erroreGrado();
}

string pura::risolvi() const{
    polinomio temp = unifica();
    temp.ordina();
    temp.eliminaZeri();

    if(temp.getMonomio_pos(1).getCoefficiente().getNumeratore()==0 || temp.size()==1)
        return "x = 0";
    razionale r = razionale(-1,1)*temp.getMonomio_pos(0).getCoefficiente()/temp.getMonomio_pos(1).getCoefficiente();

    r = razionale::potenza(r,razionale(1,2));
    return "x = +"+r.toString()+"  &&  x= -"+r.toString();
}

tipoEquazione pura::getTipo() const{
    return tipoEquazione(2);
}

razionale pura::getDelta() const{
    razionale a,c;
    polinomio temp = unifica();
    temp.ordina();
    temp.eliminaZeri();

    a=temp.getMonomio_pos(1).getCoefficiente();
    c=temp.getMonomio_pos(0).getCoefficiente();
    return  razionale(-4)*a*c;
}

File: Java/pura.java
public class pura extends secondoGrado{
    
    public pura(polinomio sx, polinomio dx) throws erroreGrado{
        super(sx,dx); 
        polinomio temp = unifica();
        temp.ordina();
        if(temp.getMonomio_pos(0).getEsponente().diverso(new razionale()))
            throw new erroreGrado();
        if(temp.getMonomio_pos(1).getEsponente().diverso(new razionale(2)))
            throw new erroreGrado();
    }

    public String getRisultato(){
        polinomio temp = unifica();
        temp.ordina();
        if(temp.getMonomio_pos(1).getCoefficiente().getNumeratore()==0)
            return "x = 0";
        razionale r = new razionale();
        razionale menoUno = new razionale(-1);
        try{
        razionale radice = new razionale(1,2);
            r =  razionale.moltiplicazioneRazionale(menoUno,razionale.divisioneRazionale(temp.getMonomio_pos(0).getCoefficiente(),temp.getMonomio_pos(1).getCoefficiente()));
            r = razionale.potenza(r,radice);
        }
        catch(Exception e){}
        return "x= +"+r.toString()+"  &&  x= -"+r.toString();
    }

    public String getTipo(){
        return "Equazione pura di secondo grado";
    }

    public razionale getDelta(){
        razionale a = new razionale();
        razionale c = new razionale();
        razionale d = new razionale(4);
        polinomio temp = unifica();
        temp.ordina();
        a=temp.getMonomio_pos(1).getCoefficiente();
        c=temp.getMonomio_pos(0).getCoefficiente();
        try{
            return razionale.sottrRazionale(new razionale(-1),razionale.moltiplicazioneRazionale(razionale.moltiplicazioneRazionale(d,a),c));
        }
        catch(Exception e){}
        return a;
    }

}